Makefile

```diff
@@ -154,10 +154,12 @@ build-rpm:
 
 # ######################################
 copy-pkgs:
-	-chmod a+rwx ../rpmbuild/RPMS/x86_64/couchdb-* ../couchdb/couchdb_* 2>/dev/null
-	-mkdir -p pkgs/couch/$(DIST) && chmod 777 pkgs/couch/$(DIST)
-	-cp ../rpmbuild/RPMS/x86_64/couchdb-* pkgs/couch/$(DIST) 2>/dev/null
-	-cp ../couchdb/couchdb_* pkgs/couch/$(DIST) 2>/dev/null
+	if [ "$(shell arch)" = "x84_64" ]; then                                                 \
```

Comment:
Can't you make this more generic rather than putting the entire 4 lines into an if statement? Perhaps something like:

```
copy-pkgs: ARCH=$(shell arch)
copy-pkgs:
        chmod a+rwx ../rpmbuild/RPMS/$(ARCH)/couchdb-* ../couchdb/couchdb_@ 2>/dev/null
        # etc etc etc
```

Reply:
It does both. `../rpmbuild/RPMS` is for .rpm (yum) builds, and `../couchdb/couchdb_*` is for .deb (dpkg) builds. I DRY'ed it out because I didn't want cut & pasted content in 2 targets. Make sense?

Reply:
Yes it makes sense, but unfortunately it does not work.

Currently (even with the ARCH fix applied) we get:

```
chmod -f a+rwx ../rpmbuild/RPMS/arm64/couchdb-* ../couchdb/couchdb_* 2>/dev/null
Makefile:158: recipe for target 'copy-pkgs' failed
mkdir -p pkgs/couch/ubuntu-xenial && chmod 777 pkgs/couch/ubuntu-xenial
make: [copy-pkgs] Error 1 (ignored)
mkdir: cannot create directory 'pkgs/couch/ubuntu-xenial': Permission denied
Makefile:158: recipe for target 'copy-pkgs' failed
cp ../rpmbuild/RPMS/arm64/couchdb-* pkgs/couch/ubuntu-xenial 2>/dev/null
make: [copy-pkgs] Error 1 (ignored)
Makefile:158: recipe for target 'copy-pkgs' failed
cp ../couchdb/couchdb_* pkgs/couch/ubuntu-xenial 2>/dev/null
make: [copy-pkgs] Error 1 (ignored)
make: [copy-pkgs] Error 1 (ignored)
Makefile:158: recipe for target 'copy-pkgs' failed
```

Which does not look like a successful run and will (at the very least) confuse people.

To rub salt into the wounds, this run did actually fail for real, but this is masked by the ignored failures.  Since we change to user to `jenkins` for the build, most of the file manipulation fail due to lack of permissions.  The default mode of `pkgs/couch` et. al is `rwxrwxr-x` and is owned by the current user `1000` in my case, where as user `jenkins` has a different UID, thus most of the above fails for real.

I will submit a fixed-up patch which what I think it a reasonable (DRY) solution.

Reply:
Our successful runs def. have the "Error 1 (ignored)" in them - success is predicated on the presence of the package file in the output directory outside of the Docker container.  The chmod is to ensure we can actually modify the files outside of the build directory...

Happy to see an improvement, looking forward to your changes.

Reply:
Just to clarify - the package builds run in a container that will *definitely* have a different uid, but since they are building the package, they are the ones that build the files under the two directories being `chmod`ed. One of the two won't be found (since we are only on one path) but the other *should* succeed in its chmod and the chmod is *mandatory* otherwise we won't be able to modify/move the file outside of the container.